### PR TITLE
V22F-102 Choosing simulation information

### PR DESCRIPTION
**Pull request by shoekzema**
Added a new menu, changed the old menu. Changed the Config (now SDAConfig). Added ConfigTests.

### CODE REVIEW

**Review by KungFuDonkey — CHANGES_REQUESTED** [review]
some p_ are missing from parameters from DataSelectionMenu.cpp Researchmenu.cpp Researchmenu.h

**Comment by KungFuDonkey** [comment]
Needs to fix merge conflicts with main, you should keep in mind that main has templates now and that the templates should stay

**Review by KungFuDonkey — CHANGES_REQUESTED** [review]
Research menu should use SMediator to avoid build errors

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/SDAConfig.h [hunk 1]
```diff
@@ -3,12 +3,13 @@
...
-class Config
+class SDAConfig
 {
 private:
     INTERVENTION_TYPE m_interventionType;
     BLACK_BOX_TYPE m_blackBoxType;
     DATA_STORAGE_TYPE m_dataStorageType;
+    bool* m_dataCollectionSetting;
 
     InterventionFactory interventionFactory;
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/SDAConfig.h:14]:
```suggestion
    InterventionFactory m_interventionFactory;
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/ConfigEnums.h [hunk 2]
```diff
@@ -1,5 +1,16 @@
 #pragma once
 
+typedef unsigned int TASK;
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/ConfigEnums.h:3]:
```suggestion
typedef unsigned int Task;
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/ConfigEnums.h [hunk 3]
```diff
@@ -1,5 +1,16 @@
...
+
+#define TASK_NO_TASK        0
+#define TASK_LANE_KEEPING   1
+#define TASK_SPEED_CONTROL  2
+
+typedef unsigned int INDICATOR;
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/ConfigEnums.h:9]:
```suggestion
typedef unsigned int Indicator;
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/ConfigEnums.h [hunk 4]
```diff
@@ -8,11 +19,25 @@ typedef unsigned int INTERVENTION_TYPE;
 #define INTERVENTION_TYPE_PERFORM_WHEN_NEEDED   3
 #define INTERVENTION_TYPE_ALWAYS_INTERVENE      4
 
+typedef unsigned int PARTICIPANT_CONTROL;
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/ConfigEnums.h:22]:
```suggestion
typedef unsigned int ParticipantControl;
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/ConfigEnums.h [hunk 5]
```diff
@@ -8,11 +19,25 @@ typedef unsigned int INTERVENTION_TYPE;
...
+
+#define PARTICIPANT_CONTROL_INTERVENTIONS_ON_OFF    0
+#define PARTICIPANT_CONTROL_GAS                     1
+#define PARTICIPANT_CONTROL_STEERING                2
+
 typedef unsigned int BLACK_BOX_TYPE;
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/ConfigEnums.h:28]:
```suggestion
typedef unsigned int BlackBoxType;
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/ConfigEnums.h [hunk 6]
```diff
@@ -8,11 +19,25 @@ typedef unsigned int INTERVENTION_TYPE;
...
 
 #define BLACK_BOX_TYPE_SOCKET   0
 
 typedef unsigned int DATA_STORAGE_TYPE;
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/ConfigEnums.h:32]:
```suggestion
typedef unsigned int DataStorageType;
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/ConfigEnums.h [hunk 7]
```diff
@@ -8,11 +19,25 @@ typedef unsigned int INTERVENTION_TYPE;
...
 
 #define DATA_STORAGE_TYPE_SOCKET    0
-#define DATA_STORAGE_TYPE_SQL       1
\ No newline at end of file
+#define DATA_STORAGE_TYPE_SQL       1
+
+typedef unsigned int DATA_TO_STORE;
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/ConfigEnums.h:37]:
```suggestion
typedef unsigned int DataToStore;
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/DecisionMaker.h [hunk 8]
```diff
@@ -1,25 +1,27 @@
...
 /// @brief                 A class that can ask the black box to make a decision
 /// @tparam SocketBlackBox The SocketBlackBox type
-/// @tparam Config         The config type
-template <typename SocketBlackBox, typename Config>
+/// @tparam SDAConfig      The config type
+template <typename SocketBlackBox, typename SDAConfig>
 class DecisionMaker
 {
 public:
+    SDAConfig m_config;
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/DecisionMaker.h:15]:
```suggestion
    SDAConfig Config;
```

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/legacymenu.h [hunk 9]
```diff
@@ -165,7 +165,7 @@ class LEGACYMENU_API LegacyMenu : public GfModule, public IUserInterface
 	static bool activateMainMenu();
 
     //! SIMULATED DRIVING ASSISTANCE CHANGE: Activate the intervention menu.
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/legacymenu.h:167]: Comment is outdated

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/CHANGES.txt [hunk 10]
```diff
@@ -14,29 +14,33 @@ It is ordered by: if a file is added or changed, then if the file is in src or d
...
+        DataSelectionMenu.h
+        DataSelectionMenu.cpp
   data/
     data/
         intervention/
             CMakeLists.txt
             Intervention_Steering_Proto_256.png
             intervention.xml
-        menu/interventionMenu.xml
+        menu/ResearcherMenu.xml
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/CHANGES.txt:27]: Missing DataSelectionMenu.xml

**Review by suppergerrie2 — CHANGES_REQUESTED** [review]

**Review by sjoeman1 — CHANGES_REQUESTED** [review]
some coding standards are not adhered, and mediator.h has different indentation

**Review by suppergerrie2 — APPROVED** [review]

**Review by sjoeman1 — APPROVED** [review]

**Review by KungFuDonkey — APPROVED** [review]

**Pull request merged**